### ISSUE TITLE
Lambda Function Question Regarding Voice Command Through Alexa.

### ISSUE THREAD

**Issue by sradmard** [GRASP-ML/ServiceRobots#1]
Hi,

I have a question regarding your voice control demo through Amazon Alexa. I was wondering whether you have your Amazon Lambda Function in javascript or in Python. I am also trying to implement a similar demo, but the Alexa ROS package based on javascript seems very limited as I need to add my own message types and topics, and I cannot find any example based on Python implementation. I would really appreciate any help on this issue.
Thank you so much in advance.

**Comment by feefles** [comment]
Hi! I was one of the people who worked on the Alexa demo. So the way we did things had a few more layers than what you see in the demo. The Alexa demo was an extension of our task server functionality which isn't publicly available right now, but we essentially set up a server and a client structure. The server was a very simple storage unit for tasks: you could either add tasks or query what tasks were available. For that demo, we set up a client on the robot that was essentially a rosnode that would continuously query the server for available tasks, then execute them. 

So long story short, because of this architecture, our Alexa integration simply consisted of making a single javascript call in our Lambda to add a task to the server, and the client would take care of the rest.

**Comment by alexbaucom17** [comment]
Closing for now. Feel free to re-open if there are more questions or problems.
